prez/services/query_generation/spatial_filter.py

```diff
@@ -331,38 +331,64 @@ def generate_bbox_filter(
     if target_system in ["geosparql", "graphdb"]:  # For QLever, plain wkt is used
         processed_wkt = f"<{srid}> {wkt}"
 
-    # Triples to define the geometry path
-    tssp_list = []
-    filter_gpnts = []
+    # Create patterns to be wrapped in FILTER EXISTS
+    combined_patterns = GroupGraphPatternSub()
+    
+    # Add geometry triples
     if target_system in ["geosparql", "graphdb"]:
-        tssp_list.append(
-            TriplesSameSubjectPath.from_spo(
-                subject, IRI(value=GEO.hasGeometry), geom_bn_var
-            )
-        )
-        tssp_list.append(
-            TriplesSameSubjectPath.from_spo(geom_bn_var, IRI(value=GEO.asWKT), geom_lit_var)
+        combined_patterns.add_pattern(
+            TriplesBlock.from_tssp_list([
+                TriplesSameSubjectPath.from_spo(
+                    subject, IRI(value=GEO.hasGeometry), geom_bn_var
+                ),
+                TriplesSameSubjectPath.from_spo(
+                    geom_bn_var, IRI(value=GEO.asWKT), geom_lit_var
+                )
+            ])
         )
+        
     if target_system == "graphdb":
-        # add the filter predicate
-        tssp_list.append(
-            TriplesSameSubjectPath.from_spo(
-                geom_bn_var,  # graphdb supports the subject being a geo:Feature or geo:Geometry; geo:Geometry used here, performance not tested.
-                IRI(value=GEO.sfIntersects),
-                RDFLiteral(value=processed_wkt, datatype=IRI(value=str(GEO.wktLiteral))),
+        # Add the filter predicate triple for GraphDB
+        combined_patterns.add_pattern(
+            TriplesBlock(
+                triples=TriplesSameSubjectPath.from_spo(
+                    geom_bn_var,
+                    IRI(value=GEO.sfIntersects),
+                    RDFLiteral(value=processed_wkt, datatype=IRI(value=str(GEO.wktLiteral))),
+                )
             )
         )
 
     elif target_system in ["geosparql", "qlever"]:
-        filter_gpnts = generate_spatial_filter_clause(
+        # Add geometry triples first
+        if target_system == "geosparql":
```

Comment:
Are these same geometry triples already added above, at line 339?

Reply:
Yes - removed now

Reply:
in https://github.com/RDFLib/prez/pull/410/commits/e1cb3c77d65a48af6f63c8d1a8ced69417768e0d